openquake/hazardlib/source/base.py

```diff
@@ -40,6 +40,8 @@ class BaseSeismicSource(with_metaclass(abc.ABCMeta)):
     __slots__ = ['source_id', 'name', 'tectonic_region_type',
                  'trt_model_id', 'weight', 'seed', 'id']
 
+    MODIFICATIONS = abc.abstractproperty()
```

Comment:
As a Python programmer, if I see a line

``MODIFICATIONS = abc.abstractproperty()``

I assume that the intent of the coder was to force people to specify the class attribute ``MODIFICATIONS`` in subclasses, along the lines of the Zen of Python, "explicit is better than implicit". Using ``MODIFICATIONS = set(())``   would be preferable if this is a valid common default. i.e. if the user is not expected to specify this attribute except in specific cases. What was your intent, Graeme?